sktime/utils/tests/test_plotting.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+import pandas as pd
+import pytest
+
+from sktime.datasets import load_airline
+from sktime.utils.plotting import plot_series, plot_correlations
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+from sktime.utils.validation.series import VALID_DATA_TYPES
+
+ALLOW_NUMPY = False
+y_airline = load_airline()
+y_airline_true = y_airline.iloc[y_airline.index < "1960-01"]
+y_airline_test = y_airline.iloc[y_airline.index >= "1960-01"]
+series_to_test = [y_airline, (y_airline_true, y_airline_test)]
+
+
+def _plot_series(series, ax=None, **kwargs):
+    if isinstance(series, tuple):
+        return plot_series(*series, ax=ax, **kwargs)
+    else:
+        return plot_series(series, ax=ax, **kwargs)
+
+
+@pytest.fixture
+def valid_data_types():
+    valid_data_types = tuple(filter(lambda x: x is not np.ndarray, VALID_DATA_TYPES))
+    return valid_data_types
+
+
+@pytest.mark.parametrize("series_to_plot", series_to_test)
+def test_plot_series_runs_without_error(series_to_plot):
+    _check_soft_dependencies("matplotlib", "seaborn")
+    import matplotlib.pyplot as plt
+
+    _plot_series(series_to_plot)
+    plt.gcf().canvas.draw_idle()
+
+
+@pytest.mark.parametrize("series_to_plot", [y_airline.values, "this_is_a_string"])
+def test_plot_series_invalid_input_type_raises_error(series_to_plot, valid_data_types):
+    # TODO: Is it possible to dynamically create the matching str if it includes
+    #       characters that need to be escaped (like .)
+    # match = f"Data must be a one of {valid_data_types}, but found type: {type(Z)}"
+    with pytest.raises(TypeError):
+        _plot_series(series_to_plot)
+
+
+@pytest.mark.parametrize(
+    "series_to_plot", [(y_airline_true, y_airline_test.reset_index(drop=True))]
+)
+def test_plot_series_with_unequal_index_type_raises_error(
+    series_to_plot, valid_data_types
+):
+    match = "Found series with different index types."
+    with pytest.raises(TypeError, match=match):
+        _plot_series(series_to_plot)
+
+
+@pytest.mark.parametrize("series_to_plot", series_to_test)
+def test_plot_series_invalid_marker_kwarg_len_raises_error(series_to_plot):
+    match = """There must be one marker for each time series,
+                but found inconsistent numbers of series and
+                markers."""
+    with pytest.raises(ValueError, match=match):
+        # Generate error by creating list of markers with length that does
+        # not match input number of input series
+        if isinstance(series_to_plot, pd.Series):
+            markers = ["o", "o"]
+        elif isinstance(series_to_plot, pd.DataFrame):
```

Comment:
does it work with data frames? if so, we need to add it to the docstring for `plot_series`, but `series_to_test` doesn't seem to include any dataframe either